### DISCUSSION TITLE
workerman internals

### DISCUSSION THREAD

**Discussion by diegogit03** [walkor/Workerman#906] [Q&A] [answered]
how workerman works in internals? this is more simply amazing!

questions:

its have a event loop like nodejs event loop?
its use socket_* functions?

i am just a curious programmer, and i am trying read the code, but i not have a php vocabulary for this :D

**Comment by walkor** [comment, marked as answer]
Yes, there is an event loop inside workerman. workerman not use `sockets_*` functions but `stream_*` functions .

**Reply by diegogit03** [reply]
where i start for read the code? which file?

**Reply by walkor** [reply]
Worker.php
